### ISSUE TITLE
false-positive missing tracks remain in Missing even though files exist and can be played

### ISSUE THREAD

**Issue by ronso0** [mixxxdj/mixxx#13533]
### Bug Description

For some reason some track files have been marked missing (fs_deleted) during a scan even though files still exist and haven't been modified in any way.

After scanning the music directories again, these files remain in Missing, but I can load and play them from there.
I suspect some issue with either `needs_verification` or releated flags here
for (const auto& rootDir : libraryRootDirs) {

**Comment by LindyBalboa** [comment]
I would just like to add that this has been an issue for me as well using 2.5.0 on Windows 10.

**Comment by ronso0** [comment]
@LindyBalboa can you check your mixxxdb.sqlite -> `track_locations` table if it contains paths like `/../../`?

**Comment by LindyBalboa** [comment]
@ronso0 that was not the issue, but it led me to the issue. At some point when I moved laptops, the directory prefix of my music library changed. While the `location` field was updated to reflect this change, not every `directory` field was properly updated. After correcting `directory` to match `location`, none of those tracks end up missing after a library re-scan.

**Comment by ronso0** [comment]
Great, thanks for taking a look!

While working on another issue I couldn't figure when the `directory` field is populated.
Just now I discovered one place, called when reloacting a directory in the Library preferences:
https://github.com/mixxxdj/mixxx/blob/0b025c49214a4452f15e7d91bcb999bc0563ba6d/src/library/dao/directorydao.cpp#L252-L266

> At some point when I moved laptops, the directory prefix of my music library changed.

How did you migrate/update the library?

**Comment by ronso0** [comment]
So I guess my issue actually stems from the /../.. location part because `TrackDAO::detectMovedTracks()` fails with that prefix.

**Comment by LindyBalboa** [comment]
> Great, thanks for taking a look!
> 
...
> > At some point when I moved laptops, the directory prefix of my music library changed.
> 
> How did you migrate/update the library?

Honestly, at this point I'm not really sure. I have a vague recollection of it not happening "automagically" and needing to mess with the DB, but I couldn't say exactly how it went.

**Comment by ronso0** [comment]
Okay. Since I'm not sure when the issue occured for me (noticed the missing tracks only after quite a few session after a rescan), so I'll leave this issue open.

Now I'll notice much earlier because I have #13427 integrated into my personal branch (with the scan summary popup).
Someone may look into the symlink resolve issue.
